File: OdevBackend/service/src/main/java/com/example/service/model/Game.java
package com.example.service.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "games")
@Data
@NoArgsConstructor
public class Game {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String slug;
    private String name;
    private LocalDate released;
    private String backgroundImage;
    private Double rating;
    private Integer ratingTop;

    @ManyToMany
    @JoinTable(name = "games_platforms",
            joinColumns = @JoinColumn(name = "game_id"),
            inverseJoinColumns = @JoinColumn(name = "platform_id"))
    private List<Platform> platforms;

    @ManyToMany
    @JoinTable(name = "games_genres",
            joinColumns = @JoinColumn(name = "game_id"),
            inverseJoinColumns = @JoinColumn(name = "genre_id"))
    private List<Genre> genres;

    @ManyToMany
    @JoinTable(name = "game_prices",
            joinColumns = @JoinColumn(name = "game_id"),
            inverseJoinColumns = @JoinColumn(name = "id"))
    private List<GamePrice> prices;

    @OneToMany()
    @JoinColumn(name = "game_id")
    private List<ShortScreenshot> shortScreenshots;

    @ManyToMany
    @JoinTable(name = "comments",
            joinColumns = @JoinColumn(name = "game_id"),
            inverseJoinColumns = @JoinColumn(name = "id"))
    private List<Comment> comments;
}

File: mobil/lib/models/Game.dart
import 'Comment.dart';
import 'GamePrices.dart';
import 'Genres.dart';
import 'ShortScreenshots.dart';

class Game {
  int? id;
  String? slug;
  String? name;
  String? released;
  String? backgroundImage;
  double? rating;
  int? ratingTop;
  List<ShortScreenshots> shortScreenshots = [];
  List<Genres> genres = [];
  List<GamePrices> prices = [];
  List<Comments> comments = [];

  Game(
      this.id,
      this.slug,
      this.name,
      this.released,
      this.backgroundImage,
      this.rating,
      this.ratingTop,
      this.shortScreenshots,
      this.genres,
      this.comments);

  Game.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    slug = json['slug'];
    name = json['name'];
    released = json['released'];
    backgroundImage = json['backgroundImage'];
    rating = json['rating'];
    ratingTop = json['rating_top'];
    if (json['shortScreenshots'] != null) {
      shortScreenshots = [];
      json['shortScreenshots'].forEach((v) {
        shortScreenshots.add(ShortScreenshots.fromJson(v));
      });
    }
    if (json['genres'] != null) {
      genres = [];
      json['genres'].forEach((v) {
        genres.add(Genres.fromJson(v));
      });
    }
    if (json['prices'] != null) {
      prices = [];
      json['prices'].forEach((v) {
        prices.add(GamePrices.fromJson(v));
      });
    }
    if (json['comments'] != null) {
      comments = [];
      json['comments'].forEach((v) {
        comments.add(Comments.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['slug'] = slug;
    data['name'] = name;
    data['released'] = released;
    data['background_image'] = backgroundImage;
    data['rating'] = rating;
    data['rating_top'] = ratingTop;
    data['short_screenshots'] = shortScreenshots.map((v) => v.toJson()).toList();
    data['genres'] = genres.map((v) => v.toJson()).toList();
    return data;
  }
}


